File: Cpp/Longest Increasing Subsequence.cpp
class Solution {
public:
    /**
     * @param nums: The integer array
     * @return: The length of LIS (longest increasing subsequence)
     */
    int longestIncreasingSubsequence(vector<int> nums) {
        // write your code here
        vector<int> vec(nums.size(), 1);
        for (int i = 1; i < nums.size(); i ++) {
            for (int j = 0; j < i; j ++) {
                if (nums[i] >= nums[j]) {
                    vec[i] = max(vec[i], vec[j]+1);
                }
            }
        }
        int cnt = 0;
        for (int i = 0; i < vec.size(); i ++) {
            cnt = max(cnt, vec[i]);
        }
        return cnt;
    }
};

File: Java/Longest-Increasing-Subsequence.java
public class Solution {
    /**
     * @param nums: The integer array
     * @return: The length of LIS (longest increasing subsequence)
     */
    public int longestIncreasingSubsequence(int[] nums) {
        // write your code here
        int len = nums.length;
        int []dp = new int[len];
        for (int i = 0; i < len; i ++) {
        	dp[i] = 1;
        	for (int j = 0; j < i; j ++) {
        		if (nums[i] >= nums[j]) {
        			dp[i] = Math.max(dp[i], dp[j] + 1);
        		}
        	}
        }
        int maxlen = 0;
        for (int i = 0; i < len; i ++) {
        	maxlen = Math.max(maxlen, dp[i]);
        }
        return maxlen;
    }
}

